legacy/application/models/StreamSetting.php

```diff
@@ -2,70 +2,94 @@
+    {
+        return intval(trim($key, 's'));
+    }
 
-        if ($result == 1) {
-            $sql = 'UPDATE cc_stream_setting'
-                . ' SET value = :value, type = :type'
-                . ' WHERE keyname = :key';
-        } else {
-            $sql = 'INSERT INTO cc_stream_setting (keyname, value, type)'
-                . ' VALUES (:key, :value, :type)';
+    public static function getOutput($key, $add_prefix = false)
+    {
+        $id = self::toOutputId($key);
+        $config_id = $id - 1;
+        $prefix = $add_prefix ? "s{$id}_" : '';
+
+        if (!Config::has("stream.outputs.{$config_id}")) {
+            return [
+                $prefix . 'enable' => 'false',
+                $prefix . 'output' => 'icecast',
+                $prefix . 'host' => 'localhost',
+                $prefix . 'port' => 8000,
+                $prefix . 'mount' => '',
+                $prefix . 'user' => 'source',
+                $prefix . 'pass' => '',
+                $prefix . 'admin_user' => 'admin',
+                $prefix . 'admin_pass' => '',
+                $prefix . 'channels' => 'stereo',
+                $prefix . 'bitrate' => 128,
+                $prefix . 'type' => '',
+                $prefix . 'name' => '',
+                $prefix . 'description' => '',
+                $prefix . 'genre' => '',
+                $prefix . 'url' => '',
+                $prefix . 'mobile' => 'false',
+            ];
         }
 
-        $stmt = $con->prepare($sql);
-        $stmt->bindParam(':key', $key);
-        $stmt->bindParam(':value', $value);
-        $stmt->bindParam(':type', $type);
-
-        if ($stmt->execute()) {
-            // do nothing
-        } else {
-            $msg = implode(',', $stmt->errorInfo());
+        $output = Config::get("stream.outputs.{$config_id}");
 
-            throw new Exception("Error: {$msg}");
-        }
+        return [
+            $prefix . 'enable' => $output['enabled'] ?? 'false',
+            $prefix . 'output' => $output['kind'] ?? 'icecast',
+            $prefix . 'host' => $output['host'] ?? 'localhost',
+            $prefix . 'port' => $output['port'] ?? 8000,
+            $prefix . 'mount' => $output['mount'],
+            $prefix . 'user' => $output['source_user'] ?? 'source',
+            $prefix . 'pass' => $output['source_password'] ?? '',
+            $prefix . 'admin_user' => $output['admin_user'] ?? 'admin',
+            $prefix . 'admin_pass' => $output['admin_password'] ?? '',
+            $prefix . 'channels' => $output['audio']['channels'] ?? 'stereo',
+            $prefix . 'bitrate' => $output['audio']['bitrate'] ?? 128,
+            $prefix . 'type' => $output['audio']['format'],
+            $prefix . 'name' => $output['name'] ?? '',
+            $prefix . 'description' => $output['description'] ?? '',
+            $prefix . 'genre' => $output['genre'] ?? '',
+            $prefix . 'url' => $output['url'] ?? '',
+            $prefix . 'mobile' => 'false',
```

Comment:
This will always disable a feature that I am unaware off, I could use some input to know what this is really for and if we want to keep it !

Reply:
No idea what that is...

Reply:
Ok, lets just leave it disabled for now.